Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -277,8 +311,35 @@ class MessageUpdater: Worker {
         messageId: MessageId,
         completion: ((Error?) -> Void)? = nil
     ) {
-        apiClient.request(endpoint: .deleteReaction(type, messageId: messageId)) {
-            completion?($0.error)
+        var reaction: MessageReactionDTO?
+
+        database.write { session in
+            do {
+                reaction = try session.removeReaction(from: messageId, type: type, on: nil)
+            } catch {
+                log.warning("Failed to remove the reaction from to the database: \(error)")
+            }
+
+            guard let reaction = reaction else {
+                return
+            }
+            reaction.localState = .pendingDelete
+        } completion: { error in
```

Comment:
this db error should probably also be respected 🤔 e.g. if it's non-nil we should trigger the completion and stop

Reply:
this is assuming that the database is more important than the API call and if db errors for some reason user input should fail. I think we're better off giving up on the whole optimistic UI part in case of db error and fallback to what we usually do (perform an API call and rely on event handling to update the message)